core/src/main/java/io/kestra/core/test/flow/UnitTestResult.java

```diff
@@ -0,0 +1,29 @@
+package io.kestra.core.test.flow;
+
+
+import jakarta.validation.constraints.NotNull;
+
+import java.util.List;
+
+public record UnitTestResult(
+    @NotNull
+    String unitTestId,
+    @NotNull
+    String unitTestType,
+    @NotNull
+    Status status,
+    @NotNull
+    List<AssertionResult> assertionResults
+) {
+
+    public static UnitTestResult of(String unitTestId, String unitTestType, List<AssertionResult> results) {
+        var status = results.stream().anyMatch(assertion -> !assertion.isSuccess()) ? Status.FAILED : Status.SUCCESS;
+        return new UnitTestResult(unitTestId, unitTestType, status, results);
+    }
+
+    public enum Status {
+        SUCCESS,
+        FAILED,
+        SKIPPED
+    }
```

Comment:
Maybe use the same enum for UnitTestResult and AssertionResult to avoid duplication?